src/evaluators/BinaryExpression.js

```diff
@@ -195,6 +197,17 @@ export function computeBinary(
   let resultType;
   const compute = () => {
     if (lval instanceof AbstractValue || rval instanceof AbstractValue) {
+      // If the left-hand side of an instanceof operation is a primitive,
+      // and the right-hand side is a simple object (it does not have [Symbol.hasInstance]),
+      // then the result should always compute to `false`.
+      if (
+        op === "instanceof" &&
+        lval instanceof PrimitiveValue &&
+        (rval instanceof ObjectValue || rval instanceof AbstractObjectValue) &&
```

Comment:
The `rval instanceof ObjectValue` is unnecessary since one of them needs to be an abstract value.

You could expand the `lval` to include abstract values with `Value.isTypeCompatibleWith(lval.getType(), PrimitiveValue)`. That way you handle the abstract primitive case too.

https://github.com/facebook/prepack/blob/master/src/methods/to.js#L682